src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1736,12 +1736,30 @@ private function addRateLimiterSection(ArrayNodeDefinition $rootNode)
                             ->useAttributeAsKey('name')
                             ->arrayPrototype()
                                 ->children()
-                                    ->scalarNode('lock')->defaultValue('lock.factory')->end()
-                                    ->scalarNode('storage')->defaultValue('cache.app')->end()
-                                    ->scalarNode('strategy')->isRequired()->end()
-                                    ->integerNode('limit')->isRequired()->end()
-                                    ->scalarNode('interval')->end()
+                                    ->scalarNode('lock_factory')
+                                        ->info('The service ID of the lock factory used by this limiter')
+                                        ->defaultValue('lock.factory')
+                                    ->end()
+                                    ->scalarNode('cache_pool')
+                                        ->info('The cache pool to use for storing the current limiter state')
+                                        ->defaultValue('cache.app')
+                                    ->end()
+                                    ->scalarNode('storage_service')
+                                        ->info('The service ID of a custom storage implementation, this precedes any configured "cache_pool"')
+                                        ->defaultNull()
+                                    ->end()
+                                    ->enumNode('strategy')
+                                        ->info('The rate limiting algorithm to use for this rate')
+                                        ->isRequired()
+                                        ->values(['fixed_window', 'token_bucket'])
+                                    ->end()
+                                    ->integerNode('limit')
+                                        ->info('The maximum allowed hits in a fixed interval or burst')
+                                        ->isRequired()
+                                    ->end()
+                                    ->scalarNode('interval')->info('Configures the fixed interval if "strategy" is set to "fixed_window"')->end()
```

Comment:
Could we modify this -> `Configures the fixed interval` to mention if it's in minutes, seconds, ...? Example:

```
Configures the fixed interval in seconds if "strategy" is set to "fixed_window"
```

Reply:
Hmm, I agree that we have to mention to format. However, I'm open for suggestions on how to briefly describe this. It's a human readable time string. More precisely it's a number followed by any `unit` documented in https://www.php.net/datetime.formats.relative

Reply:
Linking there would be good. Adding an example also would be super great - I'm still confused after following that link.

Reply:
Hmm, I've now made it a bit more simple and explicit (more syntax is allowed, but this is probably the most used):

> The value must be a number followed by "second", "minute", "hour", "day", "week" or "month" (or their plural equivalent)